agent/eventhandler/handler.go

```diff
@@ -28,6 +28,8 @@ var statesaver statemanager.Saver = statemanager.NewNoopStateManager()
 
 func HandleEngineEvents(taskEngine engine.TaskEngine, client api.ECSClient, saver statemanager.Saver) {
 	statesaver = saver
+	handler := NewTaskHandler()
```

Comment:
It looks like now you have two separate `taskHandler`s: one here and one in `agent/acs/handler/payload_handler.go`.  Is that intentional?